File: robot-plugin/src/main/java/amailp/file/RobotFileType.java
package amailp.file;

import amailp.language.RobotLanguage;
import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class RobotFileType extends LanguageFileType {
    public static final RobotFileType INSTANCE = new RobotFileType();

    private RobotFileType() {
        super(RobotLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Robot file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Robotframework language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "robot";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return Icons.FILE;
    }
}

File: src/main/scala/amailp/intellij/robot/idea/RobotFileType.scala
package amailp.intellij.robot.idea

import com.intellij.openapi.fileTypes.{FileTypeFactory, LanguageFileType, FileTypeConsumer}
import javax.swing._
import com.intellij.openapi.util.IconLoader

object Icons {
  val file = IconLoader.getIcon("/icons/robot_file_16.png")
}

object RobotFileType extends LanguageFileType(RobotLanguage) {
  val getName: String = "Robot file"
  val getDescription: String = "Robotframework language file"
  val getDefaultExtension: String = "robot"
  val getIcon: Icon = Icons.file
}

class RobotFileTypeFactory extends FileTypeFactory {
   def createFileTypes (consumer: FileTypeConsumer) = consumer.consume(RobotFileType, "robot")
}


